File: src/main/java/com/erisu/cloud/megumi/event/annotation/Event.java
package com.erisu.cloud.megumi.event.annotation;

import org.springframework.stereotype.Component;

import java.lang.annotation.*;

/**
 * 事件注册工具
 */
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Component
@Documented
public @interface Event {
}

File: megumi/src/main/java/com/erisu/cloud/megumi/event/Event.kt
package com.erisu.cloud.megumi.event

import org.springframework.stereotype.Component

/**
 * 事件注册工具
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Component
@MustBeDocumented
annotation class Event
